pl_bolts/utils/arguments.py

```diff
@@ -0,0 +1,118 @@
+import inspect
+from argparse import ArgumentParser, Namespace
+from dataclasses import dataclass
+from typing import Any, Optional
+
+import pytorch_lightning as pl
+
+
+@dataclass(frozen=True)
+class LitArg:
+    """Dataclass to represent init args of an object
+    """
+    name: str
+    types: tuple
+    default: Any
+    required: bool = False
+    context: Optional[str] = None
+
+
+class LightningArgumentParser(ArgumentParser):
```

Comment:
Maybe add a docstring? Have an example of that shows the usage:
```
parser = LightningArgumentParser()
parser.add_object_args("data", MyDataModule)
parser.add_object_args("model", MyModel)
args = parser.parse_args()
```
doesn't need to be in docstring necessarily, could be in docs.